### PR TITLE
Add support for custom links

### PR DESCRIPTION
**Pull request by uklotzde**

### CODE REVIEW

**Comment by uklotzde** [comment]
@navid-kalaei Does the extended API with url/title/description for custom links work for you?

**Comment by navid-kalaei** [comment]
@uklotzde Thank you so much for your quick consideration Uwe. I read the raw codes of Rust from the commits, but I was wondering if you could be so nice to provide an example in any kind examples like a sample request or even better updating the swagger documentation, so I would be able to integrate the front end to the awesome work of your API.

I could not find the documentation on swagger, if I was not careful enough to read the documentation more carefully, could I kindly ask you to guide me through the right model that should be sent to the API?

Thank you so much for your warm support :rose:

**Comment by flosse** [comment]
> or even better updating the swagger documentation

@navid-kalaei the `openapi.yaml` was updated in this PR.
You can directly load the full updated doc by loading the file into the swagger editor:

    https://raw.githubusercontent.com/kartevonmorgen/openfairdb/094057464c57e503a2f606f9f92383f0b324f62d/openapi.yaml

**Review by flosse — APPROVED** [review]
LGTM!

File: src/ports/web/api/tests.rs [hunk 1]
```diff
@@ -91,21 +91,34 @@ fn create_place_with_tag_duplicates() {
...
+        .unwrap()
+        .all_places()
+        .unwrap()
+        .into_iter()
+        .next()
+        .unwrap();
+    assert_eq!(&place.tags[..], &["bar", "foo"]);
+    // The "https://www." prefix should be added implicitly by auto-completion!
+    assert_eq!(
+        &place.links.unwrap().custom[..],
```
Comment (navid-kalaei) [inline src/ports/web/api/tests.rs:115]: dear @uklotzde 

the field name of custom links for posting data is `links`, however, the name of the same field on the get request is `custom`. this inconsistency leads to confusion. could I kindly ask you to fix that at your convenience? and when that so notify me so I would be able to integrate front end properly?

thank you so much for being such a great developer

**Comment by uklotzde** [comment]
Internally all links are stored in a common `struct` while for the external API this structure is flattened and the internal `custom` fields property becomes `links`. In the API the other, dedicated links are prefixed, e.g. with `img_`.

No contradiction, only a simpel data mapping required for backward compatibility. The internal representation may differ to any extent from the outer API representation for whatever reason.

**Pull request merged**